### DISCUSSION TITLE
Confused on what port to use with Express

### DISCUSSION THREAD

**Discussion by ghost** [geckosio/geckos.io#131] [Help / Questions]
So does the server.listen(3000) actually start the geckos server listening on port 3000 instead of the default 9208 port? Or do I have to also add a ```io.listen()``` in there? And that also sets the express server port to 3000 and that's where the page will be served from. I must be missing something here... 
```
io.addServer(server)
server.listen(3000)
```

**Comment by ghost** [comment]
After some tinkering around I see that it does set the geckos port to 3000 also, so there's no need to do a ```io.listen()``` etc. So I'm assuming there's no issue with serving the web page and also using the same port for geckos...

One thing that I did get working that I think is pretty cool though; I changed the extension on the client to ```.mjs``` and added ```export``` to the front of it, so now I can import it directly into my client.js file with a ```import { geckos } from '/js/geckos.io-client.2.0.0.min.mjs';```. Works a treat. Then I can do a proper ``` const channel = geckos({ port: 3000 })```. Saves an extra call in the html file.

**Reply by yandeu** [reply]
Yes, you can serve your files and geckos from the same port.

👍

**Reply by ghost** [reply]
Is there any advantage to doing ```io.addServer(server)``` as opposed to ```io.listen()``` or is it just semantics? I'm assuming it's just semantics so it's really not an issue. Was just making sure I wasn't missing an optimization. Thanks for all your hard work on geckos.io. It's working great!

**Comment by ghost** [comment]
I'm sending ArrayBufferView through raw.emit. Is this faster than the regular emit?
